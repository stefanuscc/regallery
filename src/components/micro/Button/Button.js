import styles from './Button.module.scss'

export const Button = (props) => {
    const {type = 'button', isIcon=false, isActive=false, className="", onClick, ...attrs} = props
    const classNames = `${styles.button} ${isIcon ? styles.icon : ''} ${isActive ? styles.active : ''} ${className}`

    return (
        <button {...attrs} type={type} className={classNames} onClick={(event) => { onClick(event) }}></button>
    )
}

export default Button