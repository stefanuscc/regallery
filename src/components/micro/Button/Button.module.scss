// Vars
@import "/src/_vars";

.button {
    border: 1px solid;
    overflow: hidden;
    outline: none;

    align-items: center;
    background-color: rgb(255 255 255);
    border-color: rgb(209 213 219);
    border-width: 1px;
    color: rgb(55 65 81);
    display: inline-flex;
    font-size: .875rem;
    font-weight: 500;
    line-height: 1.25rem;
    padding: 0.5rem 1rem;
    position: relative;
    cursor: pointer;

    &.active {
        background-color: $secondary !important;
        border-color: $secondary;
        color: white;
    }

    &.icon {
        padding: 0.5rem 0.7rem;
    }
}