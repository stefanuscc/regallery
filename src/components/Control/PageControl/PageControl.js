import styles from './PageControl.module.scss'

import { useCallback } from 'react'
import Button from '../../micro/Button/Button'

export const PageControl = (props) => {
    const {page, max, className, onChange} = props

    // Handle next page, page over than max will not count
    const handleNext = useCallback(() => {
        let newPage = page + 1 > max ? max : page + 1
        onChange(newPage)
    }, [page, max, onChange])

    // Handle prev page, page bellow than 1 will not count
    const handlePrev = useCallback(() => {
        let newPage = page - 1 < 1 ? 1 : page - 1
        onChange(newPage)
    }, [page, onChange])

    // Handle change page
    const handlePage = useCallback((newPage) => {
        onChange(newPage)
    }, [onChange])

    return (
        <span className={`${styles.group} ${className}`}>
            <Button isIcon={true} onClick={handlePrev} className="">
                <span className="sr-only">Previous</span>
                <svg xmlns="http://www.w3.org/2000/svg" className="icon" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
                    <path stroke-linecap="round" stroke-linejoin="round" d="M15 19l-7-7 7-7" />
                </svg>
            </Button>
            { [...Array(max)].map((el, idx) => (
                
                <Button 
                    key={idx}
                    isActive={page === idx + 1}
                    className=""
                    disabled={page === idx + 1}
                    onClick={ () => handlePage(idx + 1) }
                >
                    <span className="sr-only">Page {idx + 1}</span>
                    {idx + 1}
                </Button>
            ))}
            <Button isIcon={true} onClick={handleNext} className="">
                <span className="sr-only">Next</span>
                <svg xmlns="http://www.w3.org/2000/svg" className="icon" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2">
                    <path stroke-linecap="round" stroke-linejoin="round" d="M9 5l7 7-7 7" />
                </svg>
            </Button>
        </span>
    )
}

export default PageControl
