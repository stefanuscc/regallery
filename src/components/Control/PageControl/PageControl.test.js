import React from 'react';
import renderer from 'react-test-renderer';
import PageControl from './PageControl';
import "../../../setupTests"
import { shallow } from 'enzyme';
import Button from '../../micro/Button/Button';

test('Page Control to match snapshot', () => {
    const component = renderer.create(
        <PageControl page={1} max={3} onChange={() => {}} className=""/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Page number is according to max', () => {
    const max = 3;
    const control = shallow(
        <PageControl page={1} max={max} onChange={() => {}} className=""/>
    );

    expect(control.find(Button)).toHaveLength(max + 2);
});