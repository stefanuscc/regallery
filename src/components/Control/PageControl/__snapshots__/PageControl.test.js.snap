// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Control to match snapshot 1`] = `
<span
  className="group "
>
  <button
    className="button icon  "
    onClick={[Function]}
    type="button"
  >
    <span
      className="sr-only"
    >
      Previous
    </span>
    <svg
      className="icon"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15 19l-7-7 7-7"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </button>
  <button
    className="button  active "
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <span
      className="sr-only"
    >
      Page 
      1
    </span>
    1
  </button>
  <button
    className="button   "
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="sr-only"
    >
      Page 
      2
    </span>
    2
  </button>
  <button
    className="button   "
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span
      className="sr-only"
    >
      Page 
      3
    </span>
    3
  </button>
  <button
    className="button icon  "
    onClick={[Function]}
    type="button"
  >
    <span
      className="sr-only"
    >
      Next
    </span>
    <svg
      className="icon"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 5l7 7-7 7"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </button>
</span>
`;
